assets/js/admin/woocommerce_admin.js

```diff
@@ -73,6 +83,19 @@ jQuery( function ( $ ) {
 			}
 		})
 
+		.on( 'keyup', '.wc_input_strict_price[type=text]', function() {
+			var regex    = new RegExp( '[^0-9\\' + woocommerce_admin.mon_decimal_point + ']+', 'gi' );
```

Comment:
Sorry, you're right, I was testing with a regex tool the expression `[^0-9\\.]+`, which is why I got different results. But I noticed another thing, you can write a value such as this: '....' (only dots).